src/pybind/mgr/cephadm/module.py

```diff
@@ -804,6 +804,27 @@ def _get_pub_key(self):
     def _get_user(self):
         return 0, self.ssh_user, ''
 
+    @orchestrator._cli_read_command(
+        'cephadm set-user',
+        'name=user,type=CephString',
+        'Set user for SSHing to cluster hosts, passwordless sudo will be needed for non-root users')
+    def set_ssh_user(self, user):
+        current_user = self.ssh_user
+        self.set_store('ssh_user', user)
+        self._reconfig_ssh()
+        for host in self.cache.get_hosts():
```

Comment:
on one hand this violates https://docs.ceph.com/docs/master/dev/cephadm/#note-regarding-network-calls-from-cli-handlers, rendering cephadm completely unresponsive without any way to abort this operation, if we're running into timeouts. 

on the other hand, this is a potentially very problematic operations. 

I'm torn here. Maybe checking only one host is ok? I don't konw